### PR TITLE
cleaned up render_reports.R to quit() using unix error codes so scripts can call and check for success

### PR DESCRIPTION
**Pull request by matthewcornell**
@Martha and I cleaned up the file: removed commented-out code, added a check for required ensemble files, and added `quit()` calls with appropriate unix error codes.

### CODE REVIEW

File: code/reports/render_reports.R [hunk 1]
```diff
@@ -7,204 +7,132 @@ library(tidyverse)
...
 print(rmarkdown::pandoc_version())
 
+
+#
+# check for required ensemble files
+#
+
+next_saturday <- as.Date(covidHubUtils::calc_target_week_end_date(lubridate::today(), horizon = 0))
+this_monday <- next_saturday - 5
+this_monday <- as.Date("1962-10-22") # temp: use fixed date
```
Comment (nickreich) [inline code/reports/render_reports.R:26]: should this be here?

File: code/reports/render_reports.R [hunk 2]
```diff
@@ -7,204 +7,132 @@ library(tidyverse)
...
+weights_files <- c(paste0("../../ensemble-metadata/", this_monday, "-inc_death-model-weights.csv"),
```
Comment (nickreich) [inline code/reports/render_reports.R:27]: why are we looking for weights files and not the ensemble file itself?

**Review by nickreich — COMMENTED** [review]
Just a few minor comments here.

File: code/reports/render_reports.R (hunk 1 above)
Comment (matthewcornell) [reply code/reports/render_reports.R:26]: nope. thanks!

File: code/reports/render_reports.R (hunk 2 above)
Comment (matthewcornell) [reply code/reports/render_reports.R:27]: We based it on what seems to be the inputs to https://github.com/reichlab/covid19-forecast-hub/blob/master/code/reports/all-states-weekly-report.Rmd : `ensemble-metadata` files. It doesn't look like the ensemble file itself is being used for the report, right? However, if you're saying we can simplify our check by looking for one file (assuming that its presence means the `ensemble-metadata` will also be present), then great. Would you mind suggesting a snippet of code to replace it? Thanks!

Comment (nickreich) [reply code/reports/render_reports.R:27]: The ensemble file itself is not being used for the report, but the data in the file is being used, via a call to Zoltar. Those metadata files are pushed in in the same PRs as the ensemble data file itself. I think the code you suggest above Matt is better (more direct) than checking for the presence of the metadata. That said, technically, the file could be in the repo but the zoltar action might have failed so the presence of the file is not necessarily a perfect condition (unless we switched the `load_forecasts()` call to pull from the local repo) of whether the reports will run or not.

Comment (matthewcornell) [reply code/reports/render_reports.R:27]: So should I stick with the metatadata check, or go with the direct ensemble file check?

**Comment by matthewcornell** [comment]
I've updated the file to reflect the above conversation. Please have a look. Then I can run it on the vm tomorrow to see how it's working.

**Pull request merged**